src/tool_homedir.c

```diff
@@ -45,14 +52,50 @@ static char *GetEnv(const char *variable)
 }
 
 /* return the home directory of the current user as an allocated string */
-char *homedir(void)
+
+/*
+ * The original logic found a home dir to use (by checking a range of
+ * environment variables and last using getpwuid) and returned that for the
+ * parent to use.
+ *
+ * With the XDG_CONFIG_HOME support (added much later than the other), this
+ * variable is treated differently in order to not ruin existing installations
+ * even if this environment variable is set. If this variable is set, and a
+ * file name is set to check, then only if that file name exists in that
+ * directory will it be returned as a "home directory".
+ *
+ * 1. use CURL_HOME if set
+ * 2. use XDG_CONFIG_HOME if set and fname is present
+ * 3. use HOME if set
+ * 4. Non-windows: use getpwuid
+ * 5. Windows: use APPDATA if set
+ * 6. Windows: use "USERPROFILE\Application Data" is set
+ */
+
+char *homedir(const char *fname)
 {
   char *home;
 
   home = GetEnv("CURL_HOME");
   if(home)
     return home;
 
+  if(fname) {
+    home = GetEnv("XDG_CONFIG_HOME");
+    if(home) {
+      char *c = curl_maprintf("%s%s", home, fname);
```

Comment:
does this always end in a slash, you would need one otherwise

Reply:
nm I see what you did, `homedir(DIR_CHAR ".curlrc")`. It would be easier to understand as `homedir(".curlrc")` then in the function you could cat it like `"%s" DIR_CHAR "%s"`

Reply:
I think you're right. I'll address.